### PR TITLE
test: Fix true/false assignment

### PR DESCRIPTION
**Pull request by PeterHuewe**
A variable with the name ck_false should have the value CK_FALSE,
not CK_TRUE!

Signed-off-by: Peter Huewe <peter.huewe@infineon.com>

### CODE REVIEW

File: test/integration/pkcs-keygen.int.c [hunk 1]
```diff
@@ -63,7 +63,7 @@ static void test_rsa_keygen_p11tool_templ(void **state) {
     CK_SESSION_HANDLE session = ti->handle;
 
     CK_BBOOL ck_true = CK_TRUE;
-    CK_BBOOL ck_false = CK_TRUE;
+    CK_BBOOL ck_false = CK_FALSE;
```
Comment (williamcroberts) [inline test/integration/pkcs-keygen.int.c]: maybe ;-)

File: test/integration/pkcs-keygen.int.c [hunk 2]
```diff
@@ -298,7 +298,7 @@ static void test_destroy(void **state) {
     CK_SESSION_HANDLE session = ti->handle;
 
     CK_BBOOL ck_true = CK_TRUE;
-    CK_BBOOL ck_false = CK_TRUE;
+    CK_BBOOL ck_false = ck_false;
```
Comment (williamcroberts) [inline test/integration/pkcs-keygen.int.c]: Looks like this is used for CKA_SENSITIVE, weird that it's causing it to fail....

Comment (PeterHuewe) [reply test/integration/pkcs-keygen.int.c]: not really weird.
The template sets CKA_SENSITIVE to CK_FALSE, which then sets CKA_ALWAYS_SENSITIVE also to false.
verify_missing_priv_attrs_common in test.c however expects CKA_ALWAYS_SENSITIVE to be true.

Comment (PeterHuewe) [reply test/integration/pkcs-keygen.int.c]: @williamcroberts  However I'm not sure what is the correct way to fix this.
Change the template to set it to CK_TRUE, (which is not equal to the p11 template?) or change verify_missing_priv_attrs_common (to check the existence of the attribute, but not the value itself, probably needed for CKA_EXTRACTABLE as well)

Comment (williamcroberts) [reply test/integration/pkcs-keygen.int.c]: > @williamcroberts However I'm not sure what is the correct way to fix this.
> Change the template to set it to CK_TRUE, (which is not equal to the p11 template?) or change verify_missing_priv_attrs_common (to check the existence of the attribute, but not the value itself, probably needed for CKA_EXTRACTABLE as well)

I want it to match the p11 template, so I would adjust the test.

Comment (PeterHuewe) [reply test/integration/pkcs-keygen.int.c]: And of course it uncovered a bug #246 :)

**Pull request merged**